### DISCUSSION TITLE
Android studio and gradle plugin version

### DISCUSSION THREAD

**Discussion by MartinH80** [OpenTracksApp/OpenTracks#1617] [General]
I'm not an experienced developer, but know some basics. I want to give it a try, see if I can build opentracks locally, make some changes, run it and probably make pull request. Unfortunately, my first attempt fails quickly.

I am running Linux on my workstation and installed latest stable Android Studio (flamingo). Copied the repo using the github plugin, opened the project:

The project is using an incompatible version (AGP 8.1.0-beta04) of the Android Gradle plugin. Latest supported version is AGP 8.0.2

Got here: https://developer.android.com/studio/releases#android_gradle_plugin_and_android_studio_compatibility

An update to the Android studio release candidate (giraffe) should work out.

Why is the project using a beta version of gradle plugin and so introducing a dependency of the android studio release candidate? Why not just stick with latest stable versions?

**Comment by MartinH80** [comment]
Using giraffe:

The project is using an incompatible preview version (AGP 8.1.0-beta04) of the Android Gradle plugin. Current compatible preview version is AGP 8.1.0-rc01.

Fortunately I could now automatically have it update the plugin and I could build the project.

**Reply by dennisguse** [reply]
Reason:  https://github.com/OpenTracksApp/OpenTracks/issues/1527

And I am also waiting until we can switch back to AGP stable again.

**Reply by MartinH80** [reply]
Thanks, that explains! I also noticed the pull request to get to the stable release.

**Comment by dennisguse** [comment]
@MartinH80 What are you planning to do?

**Reply by MartinH80** [reply]
As I'm really unexperienced as a developer, don't expect too much :-)

I was hoping to get some practice on the process by creating a pull request for the SVG in the fdroid store https://github.com/OpenTracksApp/OpenTracks/issues/1317 (and I know about related https://github.com/OpenTracksApp/OpenTracks/issues/1444)

And I hope to build something for the attitude graph, as this is what helps me personally: https://github.com/OpenTracksApp/OpenTracks/issues/1297

I was also thinking about wanting to have more analysis features like amount of kms or average speed per week/month in a bar chart. But let's start with the above, as I'm not only knowledge-limited, but also time-limited with a family and busy job.

**Reply by dennisguse** [reply]
Sounds good!
Have fun doing it :)
If you need something, ping me.
